internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -408,14 +418,21 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{RequeueAfter: shortWait}, nil
 	}
 
+	if err := r.composite.Configure(ctx, cr, comp); err != nil {
+		log.Debug(errConfigure, "error", err)
+		r.record.Event(cr, event.Warning(reasonCompose, err))
+		return reconcile.Result{RequeueAfter: shortWait}, nil
+	}
```

Comment:
Another bit of drive-by refactoring. I moved this down from where it was at line 391 because we should validate the Composition _before_ we use it to configure the XR - not after.